src/app/clusters/ota-requestor/OTARequestor.cpp

```diff
@@ -480,20 +549,12 @@ CHIP_ERROR OTARequestor::AddDefaultOtaProvider(const ProviderLocation::Type & pr
 
     ReturnErrorOnFailure(mDefaultOtaProviderList.Add(providerLocation));
 
-    // Should be removed when periodic queries is implemented
-    iterator = mDefaultOtaProviderList.Begin();
-    while (iterator.Next())
-    {
-        SetCurrentProviderLocation(iterator.GetValue());
-        break;
-    }
-
     return CHIP_NO_ERROR;
 }
 
 void OTARequestor::OnDownloadStateChanged(OTADownloader::State state, OTAChangeReasonEnum reason)
 {
-    VerifyOrReturn(mOtaRequestorDriver != nullptr);
+    VerifyOrDie(mOtaRequestorDriver != nullptr);
```

Comment:
Why is `VerifyOrDie` needed here? Seems drastic

Reply:
Because there is really no action to be taken when this is encountered. `OTARequestor` cannot operate in the absence of `OtaRequestorDriver`, there is no way to recover from this, may as well restart -- at least it will get us to a valid state.

Reply:
Ok fair enough. I guess it's not like any other Requestor functionality will be possible without the driver